pandas/_typing.py

```diff
@@ -115,7 +114,7 @@
 Ordered = Optional[bool]
 JSONSerializable = Optional[Union[PythonScalar, List, Dict]]
 Frequency = Union[str, "DateOffset"]
-Axes = Collection[Any]
+Axes = Union[AnyArrayLike, List, Dict, range]
```

Comment:
List and Dict are probably too restrictive in general? If we use this in places where we accept list-like or dict-like.

Also from https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#conventions

> avoid invariant collection types (list, dict) in argument positions, in favor of covariant types like Mapping or Sequence;

This also helps us develop robust code since mypy will report if we try to write into the passed argument.

Reply:
This is only getting applied to the `index` and `columns` argument of `pd.DataFrame`.  We can't use `Sequence`, as that would allow a string.  We document the arguments as `Index` or `array-like`.  I don't think `List` is too restrictive - the problem is that `Sequence` is too broad because of the issue with strings.  I could see removing `Dict` since we don't document that it is acceptable.